src/helpers/model/filters.ts

```diff
@@ -1,49 +1,140 @@
 // (C) 2018 GoodData Corporation
-import { VisualizationObject } from '@gooddata/typings';
+import { VisualizationInput } from '@gooddata/typings';
 import { getQualifierObject } from './utils';
 
-export const positiveAttributeFilter = (
+export function positiveAttributeFilter(
     qualifier: string,
-    inValues: string[]
-): VisualizationObject.IVisualizationObjectPositiveAttributeFilter => ({
-    positiveAttributeFilter: {
-        displayForm: getQualifierObject(qualifier),
-        in: inValues
-    }
-});
+    inValues: string[],
+    textFilter?: boolean
+): VisualizationInput.IPositiveAttributeFilter {
+    return {
+        positiveAttributeFilter: {
+            displayForm: getQualifierObject(qualifier),
+            in: inValues,
+            textFilter
+        }
+    };
+}
 
-export const negativeAttributeFilter = (
+export function negativeAttributeFilter(
     qualifier: string,
-    notInValues: string[]
-): VisualizationObject.IVisualizationObjectNegativeAttributeFilter => ({
-    negativeAttributeFilter: {
-        displayForm: getQualifierObject(qualifier),
-        notIn: notInValues
-    }
-});
+    notInValues: string[],
+    textFilter?: boolean
+): VisualizationInput.INegativeAttributeFilter {
+    return {
+        negativeAttributeFilter: {
+            displayForm: getQualifierObject(qualifier),
+            notIn: notInValues,
+            textFilter
+        }
+    };
+}
 
-export const absoluteDateFilter = (
+export function absoluteDateFilter(
     dataSet: string,
     from?: string,
     to?: string
-): VisualizationObject.IVisualizationObjectAbsoluteDateFilter => ({
-    absoluteDateFilter: {
-        dataSet: getQualifierObject(dataSet),
-        from,
-        to
-    }
-});
+): VisualizationInput.IAbsoluteDateFilter {
+    return {
+        absoluteDateFilter: {
+            dataSet: getQualifierObject(dataSet),
+            from,
+            to
+        }
+    };
+}
 
-export const relativeDateFilter = (
+export function relativeDateFilter(
     dataSet: string,
     granularity: string,
     from?: number,
     to?: number
-): VisualizationObject.IVisualizationObjectRelativeDateFilter => ({
-    relativeDateFilter: {
-        dataSet: getQualifierObject(dataSet),
-        granularity,
-        from,
-        to
+): VisualizationInput.IRelativeDateFilter {
+    return {
+        relativeDateFilter: {
+            dataSet: getQualifierObject(dataSet),
+            granularity,
+            from,
+            to
+        }
+    };
+}
+
+export class AttributeFilterBuilder {
```

Comment:
If you split the functions as suggested above, this class will imo be unnecessary. The problem I have with it, is that you cannot chain the builder methods as every single one directly returns the filter. I.e. the following does not make sense

```ts
attributeFilter('foo').inValues(someValues).notInUris(someUris);
```

This imo defeats the purpose of the builder pattern.

Reply:
yep i've done it on purpose like this; was considering multiple functions - one per attr filter variant. then concluded internally that I like chaining/fluent API better :) 

thus arrived at:

- functions as low level factories (so developer does not need to know the shape of attribute interfaces)
- the builder / fluent API to have nicer API to create variants